server/lzqgame.js

```diff
@@ -149,12 +149,60 @@ async function playerInitialBoard({ playerName, myPositions, room }) {
   }
 }
 
-async function playerMakeMove(data) {
-  if (await isPlayerTurn(data)) { // move validation function
-    data.turn += 1;
-    console.log(`Someone made a move, the turn is now ${data.turn}`);
-    console.log(`Sending back gameState on ${data.gameId}`);
-    io.sockets.in(data.gameId).emit('playerMadeMove', data);
+// TODO: this is a utility function to determine which pieces die (if any) on movement
+const pieceMovement = (board, source, target) => {
+  const sourcePiece = board[source[0]][source[1]];
+  const targetPiece = board[target[0]][target[1]];
+
+  // there is no piece at the source tile (not a valid move)
+  if (sourcePiece === null || sourcePiece.name === 'landmine' || sourcePiece.name === 'flag') {
+    return board;
+  }
+
+  // pieces are of same affiliation
+  if (targetPiece && sourcePiece.affiliation === targetPiece.affiliation) {
+    return board;
+  }
+
+  if (targetPiece && (sourcePiece.name === 'bomb' || sourcePiece.name === targetPiece.name || targetPiece.name === 'bomb' || (sourcePiece.name !== 'engineer' && targetPiece.name === 'landmine'))) {
```

Comment:
We can probably abstract this into the util function later, because the piece comparison needs to take place on both the front and backend

Reply:
Actually, im not sure how useful the comparator function would be on the FE tbh.

Reply:
It's kind of redundant since I checked it on the frontend afterward. Safe to remove but it doesn't hurt anything.